internal/deploy/yaml.go

```diff
@@ -0,0 +1,204 @@
+package deploy
+
+import (
+	"os"
+	"strings"
+
+	"github.com/shipa-corp/ketch/internal/errors"
+	"sigs.k8s.io/yaml"
+
+	ketchv1 "github.com/shipa-corp/ketch/internal/api/v1beta1"
+	"github.com/shipa-corp/ketch/internal/utils"
+)
+
+// Application represents the fields in an application.yaml file that will be
+// transitioned to a ChangeSet.
+type Application struct {
+	Version        *string    `json:"version"`
+	Type           *string    `json:"type"`
+	Name           *string    `json:"name"`
+	Image          *string    `json:"image"`
+	Framework      *string    `json:"framework"`
+	Description    *string    `json:"description"`
+	Environment    *[]string  `json:"environment"`
```

Comment:
It would be great to add an explanation why we prefer `*[]string` to `[]string`

Reply:
I switched these slices to not be pointers-to-slices.